docs/site/Inside-Loopback-Application.md

```diff
@@ -0,0 +1,79 @@
+---
+lang: en
+title: 'Inside a Loopback Application'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Inside-Loopback-Application.html
+---
+
+## The Shopping Cart Example
+
+The example application [loopback4-example-shopping](https://github.com/strongloop/loopback4-example-shopping)
+is a nice place for developers to start looking on how to use Loopback 4.
+
+Loopback 4 supports Open API specification to define API endpoints. It also builds on the connector infrastructure
+and adds service and entity composition.
+
+In the following sections we will see how the following usecases in the shopping cart example can be configured as API endpoints. Also the example uses a REDIS datasource to save shopping cart items and a mongoDB datasource to persist user and order data.
+
+| Use case              | operations         |  summary             | 
+| --------------------  | --------------------- |----------------------|
+| create a new user     | POST /users           | Basic authentication |
+| query for a user      | GET  /users/{userId}  | query by id          |
+| login as a user       | POST /users/login     | returns a jwt token  |
+| user creates a shopping cart  | PUT /shoppingCarts/{userId}          | create or reset shopping cart  |
+| user adds items to cart       | POST /shoppingCarts/{userId}/items | updates shopping cart  |
+| user submits the order | POST /users/{userId}/orders | returns order object  |
+| user's list of all orders | GET /users/{userId}/orders | returns all orders linked with user |
+
+
+## Exposing the API endpoints
+
+Routes on the api specification are first defined as [controller methods](https://loopback.io/doc/en/lb4/Controllers.html#routing-to-controllers). For instance the following "/users/login" api endpoint is configured as login method in the [UserController](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/controllers/user.controller.ts).
+
+```
+  /users.login:
+    post:
+      responses:
+        '200':
+          description: 'Token'
+          content:
+            application/json:
+              schema:
+                "token": "string"
```

Comment:
No need to have the open api spec details here.

Reply:
agreed